### DISCUSSION TITLE
Can you somehow make the function tooltip show while typing arguments?

### DISCUSSION THREAD

**Discussion by hippie68** [yegappan/lsp#507] [Q&A]
I am used from other editors that while I'm inside a function's parentheses and typing, the whole function tooltip (description, parameters etc.) pops up and stays visible. I need this feature to learn new programming languages more quickly. Can it be activated in this plugin? By default, the tooltip is only displayed while selecting a function, as seen in the screenshot:

![lsp_function_help](https://github.com/yegappan/lsp/assets/65259318/c4677197-34b5-4d88-bc9b-11cf4bf04d75)

**Comment by D029-W** [comment]
There are two approaches,
1. Using `:LspHover` command, you need to map it yourself to use it in insert mode.
2. Setting `showSignature` option. Based on the document, this might be the option you are looking for, but I did not get this one to work.

You should check out the document for `:LspHover`. The default behavior opens the document for the symbol (in your case the print function) in a pop-up window, which disappears if you move the cursor away. 

If you want the tooltip to persist, the document provides a solution by setting `hoverInPreview` to `true`. Once you set `hoverInPreview` to `true`, the command opens up a preview window on the top that lasts until you close it.

I recommend leaving `set previewheight=10` in your vimrc file, to avoid the preview window taking too much space.

Another approach is setting `echoSignature` to true. Based on the document for `:LspShowSignature`, the option should echo the signature to the command line in insertion mode when typing. However, I did not get it to work.
> `:LspShowSignature` displays the signature of the symbol (e.g. a function or method) before the cursor in a popup.
> The popup is also automatically displayed in insert mode after entering a symbol name followed by a separator (e.g. an opening parenthesis). To disable this, you can set the `showSignature` option to false in your .vimrc file: >
>
>			LspOptionsSet({'showSignature': false})
> 
> Default is true.
>
> You can get the function signature echoed in cmd-line rather than displayed in the popup if you use:
>
>			LspOptionsSet({'echoSignature': true})
>
> Default is false.
